client/src/crypto/native.rs

```diff
@@ -86,6 +86,16 @@ pub(crate) fn generate_public_key(private_key: impl AsRef<[u8]>) -> Result<Vec<u
     Ok(public_key_uint.to_bytes_be())
 }
 
+pub(crate) fn generate_public_key_for_secret_exchange(
```

Comment:
As those APIs are only useful for the secret exchange, which is only used by the GNOME implementation I don't think we should have them here and instead move them to the server side under /gnome. Especially that we have a `Key::new`

Reply:
Yeah, it make sense to move the secret exchange related new crypto impls to /gnome. But, we don't move the new functions in keys.rs right? Also, is it ok to have the server side as a dependency for the client side?

Reply:
I didn't say anything like that, the client is a library which cannot depend on the server which is a binary...

Reply:
Sorry, I miss understood. Moved secret exchange cryto impl to gnome/crypto/.